base/feature_override_unittest.cc

```diff
@@ -0,0 +1,90 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "base/feature_override.h"
+#include "base/feature_list.h"
+#include "base/logging.h"
+#include "base/test/mock_callback.h"
+#include "testing/gmock/include/gmock/gmock.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+using testing::_;
+
+namespace base {
+namespace {
+
+const Feature kTestControlEnabledFeature{"TestControlEnabledFeature",
+                                         FEATURE_ENABLED_BY_DEFAULT};
+const Feature kTestControlDisabledFeature{"TestControlDisabledFeature",
+                                          FEATURE_DISABLED_BY_DEFAULT};
+
+const Feature kTestEnabledButOverridenFeature{"TestEnabledButOverridenFeature",
+                                              FEATURE_DISABLED_BY_DEFAULT};
+const Feature kTestDisabledButOverridenFeature{
+    "TestDisabledButOverridenFeature", FEATURE_ENABLED_BY_DEFAULT};
+
+constexpr Feature kTestConstexprEnabledButOverridenFeature{
+    "TestConstexprEnabledButOverridenFeature", FEATURE_DISABLED_BY_DEFAULT};
+constexpr Feature kTestConstexprDisabledButOverridenFeature{
+    "TestConstexprDisabledButOverridenFeature", FEATURE_ENABLED_BY_DEFAULT};
+
+OVERRIDE_FEATURE_DEFAULT_STATES({{
```

Comment:
the way this macro should be used is intentional. If we put `{` or `{{` inside the macro, then clang-format wouldn't be able to properly format lines which are `#if defined(...)`. Example:
```
OVERRIDE_FEATURE_DEFAULT_STATES({
  {kReadLater, base::FEATURE_DISABLED_BY_DEFAULT},
#if defined(OS_ANDROID)
      {kReadLater2, base::FEATURE_DISABLED_BY_DEFAULT},
#endif
      {kReadLater3, base::FEATURE_DISABLED_BY_DEFAULT},
});
```